source/shadow.c

```diff
@@ -148,19 +148,19 @@ static ShadowStatus_t validateMatchTopicParameters( const char * pTopic,
  * @param[in]  thingNameLength Length of Thing Name string pointed to by pThingName.
  * @param[in]  pShadowName Shadow Name string.
  * @param[in]  shadowNameLength Length of Shadow Name string pointed to by pShadowName.
- * @param[out] pTopicBuffer Pointer to buffer for returning the topic string.
- * @param[out] pOutLength Pointer to caller-supplied memory for returning the length of the topic string.
+ * @param[in] pTopicBuffer Pointer to buffer for returning the topic string.
+ * @param[in] pOutLength Pointer to caller-supplied memory for returning the length of the topic string.
```

Comment:
This mentions that the memory is used for returning the length of the topic string. How is that an input parameter?

Reply:
Thanks. Will be handled in follow up PR for adding a CBMC proof as discussed offline.